### PR TITLE
chore(docker): try to use a distroless image

### PR DESCRIPTION
**Pull request by balzdur**
I tried to iterate locally, it seems to be working (at least, the server start and return an error since there is no backend etc...)

Let's see how it behaves on CI + staging

### CODE REVIEW

**Review by Pascal-Delange — APPROVED** [review]
👍   thanks !

File: Dockerfile [hunk 1]
```diff
@@ -22,11 +21,12 @@ RUN --mount=type=secret,id=SENTRY_AUTH_TOKEN \
...
-FROM base AS app-builder
+FROM gcr.io/distroless/nodejs22-debian12 AS app-builder
 ENV NODE_ENV=production
 ENV PORT=${PORT:-8080}
-RUN apt-get update && apt-get install -y curl
-COPY --from=build /prod/app-builder /prod/app-builder
+COPY --from=build /prod/app-builder/node_modules /prod/app-builder/node_modules
+COPY --from=build /prod/app-builder/build /prod/app-builder/build
 WORKDIR /prod/app-builder
+USER nonroot
```
[the remark this answers is not in the document]
Comment (balzdur) [reply Dockerfile:30]: Start the container with a non root user.
I've just copy paste this line from a post I looked (not sure if it's usefull, but if it works, why not 🤷🏽)

File: Dockerfile [hunk 2]
```diff
@@ -22,11 +21,12 @@ RUN --mount=type=secret,id=SENTRY_AUTH_TOKEN \
...
 EXPOSE $PORT
-CMD [ "pnpm", "start"]
+CMD ["./node_modules/@remix-run/serve/dist/cli.js", "./build/server/index.js"]
```
[the remark this answers is not in the document]
Comment (balzdur) [reply Dockerfile:32]: The distroless image is "anoying" : it takes the `CMD` and pass it to NodeJS (IIUC)

Before, we required `/bin/sh` to execute a script in which node is called on this file. I just extracted the logic from this shell file to directly call the Node command.

The second arg is passed to the first script.

[the remark this answers is not in the document]
Comment (Pascal-Delange) [reply Dockerfile:32]: yep that was my understanding
ok, smart 😅 
didn't imagine that the js script takes a js script as input

[the remark this answers is not in the document]
Comment (Pascal-Delange) [reply Dockerfile:32]: anyway, that's exactly the point where I was stuck the other day when I tried this

[the remark this answers is not in the document]
Comment (balzdur) [reply Dockerfile:32]: In short : the remix CLI is more or less an "express bundled app to serve our Remix code".

So the first script is a CLI written in JS to :
- take the server bundle path
- start an express server using pre-built setup + CLI options
- serve our app (using our bundle, it infer the server rooting)

For your curiosity, it is possible to use another server to serve Remix, `remix-serve` is provided as an easy auto-solution. You can find [here](https://github.com/kentcdodds/kentcdodds.com/blob/main/server/index.ts) an example of this (using express). The interesting part is the `app.all(*)`: it passes every requests to Remix.

**Pull request merged**